### PR TITLE
The app sub should be unsubscribed when the app sub time window is bl…

### PR DESCRIPTION
**Pull request by xiangjingli**
…ocked

Signed-off-by: Xiangjing Li <xiangli@redhat.com>

https://github.com/open-cluster-management/backlog/issues/9310

### CODE REVIEW

File: pkg/controller/subscription/subscription_controller.go [hunk 1]
```diff
@@ -369,6 +378,23 @@ func (r *ReconcileSubscription) doReconcile(instance *appv1.Subscription) error
 	}
 
 	if sub, ok := r.subscribers[subtype]; ok {
+		// If the app sub time window is blocked, un-subscribe the app sub
```
Comment (ianzhang366) [inline pkg/controller/subscription/subscription_controller.go:381]: not sure if the unsubscribe would delete the deployed resource tho... if so, is it what we want? 

I mean change from active to block, to me, means new objects won't be deployed, and old objects wouldn't be deleted...

again, if unsubscribe won't delete the old resource, then there's not a problem with this  logic

File: pkg/synchronizer/kubernetes/sync_client.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func (sync *KubeSynchronizer) AddTemplates(subType string, hostSub types.Namespa
 
 	select {
 	case sync.tplCh <- rsOrder:
-		klog.V(1).Info("wrote resource request/order to cache")
+		klog.V(1).Infof("wrote resource request/order to cache: %#v", rsOrder)
```
Comment (ianzhang366) [inline pkg/synchronizer/kubernetes/sync_client.go:125]: A minor comment,

`rsOrder` might be a giant object... print the whole object out could be too much, maybe we should add an `String()` func to `rsOrder`.

In this way, we can do
`klog.V(1).Infof("wrote resource request/order to cache: %s", rsOrder)`

Then within the string func, we can decide what would be print out for debugging or tracing

**Comment by rokej** [comment]
@xiangjingli If you unsubscribe, it will delete the application. Block means no update to the application but the application should be running. I believe this is what Git subscription does today. It looks like the bug was opened for Helm application.

**Review by rokej — CHANGES_REQUESTED** [review]
@xiangjingli If you unsubscribe, it will delete the application. Block means no update to the application but the application should be running. I believe this is what Git subscription does today. It looks like the bug was opened for Helm application.

File: pkg/synchronizer/kubernetes/sync_client.go (hunk 2 above)
Comment (xiangjingli) [reply pkg/synchronizer/kubernetes/sync_client.go:125]: yes, let me revert this change

**Comment by xiangjingli** [comment]
@ianzhang366  @rokej   The issue does happen in helm subscription.  I confirmed that in current git subscription, when time window is from active to block, the deployed resources are removed.  So the fix is to follow the git behaviour to all types of subscriptions.

I feel both you and  @thuyn-581  comments  make sense to me   :-)

So Block could mean
1.  no update to the application but the application should be running.  
or 2.   the deployed application should be removed. In next active time window, the application will be deployed again. 

@thuyn-581 , Could you comment which one you prefer to do?  Since the current git subscription does remove the deployed resources when the time window is blocked, it looks 2 is the way to go.  

Besides 1 is a bit surprise to me as well -  When I block the time window, I do expect no resources are deployed. If not, it looks the time window is not working.

File: pkg/controller/subscription/subscription_controller.go (hunk 1 above)
Comment (xiangjingli) [reply pkg/controller/subscription/subscription_controller.go]: As discussed,  when time window is blocked, any new update of the application. will be blocked. But the deployed resources should be remained.

**Pull request merged**